packages/chain/contracts/Eth_broker.sol

```diff
@@ -339,7 +338,7 @@ contract Eth_broker {
             "Curve burn failed"
         );
         // Getting expected ETH for DAI
-        uint256 ethMin = sellRewardDai(_minDaiSellValue);
+        uint256 ethMin = sellRewardDai(dai_.balanceOf(address(this)));
```

Comment:
why does this revert https://github.com/ethersphere/bzzaar-contracts/pull/7/files ?

Reply:
a bit more info - i believe the original change was made in order to allow the caller of the function to account for slippage. using the full amount of DAI received by the contract to calculate the minimum amount of ETH that would be received as a result of the transaction would only work in the case that the slippage is 0%. @ralph-pichler @RyRy79261 could you advise please guys? 😁

Reply:
More context: https://github.com/ethersphere/bzzaar-contracts/pull/7

Reply:
I would be really interested to hear Ralph has to say about this, since he approved #7 by Nicca42. Completely tracing this down to the source is probably warranted as this value might have an impact. I can help with that, but that will take a bit more time than just doing a review. 
On the most rudimentary level, the function `sellRewardDai` says that `_daiAmount` is the amount of dai to be traded for ETH and this is indeed `dai_.balanceOf(address(this))` and not `_minDaiValue`. However, I see that #7 changed  the value passed in at this line (which is now reverted), but did not update the comment for the function `sellRewardDai`.

Reply:
Veronica writes that this could not be properly tested because the solidity versions between the router and this contract are different. If you do this change, at the very least I would call for caution and manually test that this works (and share the test artifacts)

Reply:
Hi there, sorry it's been a while since I saw this code base so still need to pull it down but if I recall correctly, in trying to eliminate all potential entropy from the contracts when trying to fix the tests, I think this approach is more explicit as it takes the active balance as directly as possible 

That being said it's be a while so I can't recall and specifically deliberate requirement for this line change